Modules/gcmodule.c

```diff
@@ -305,39 +366,37 @@ visit_reachable(PyObject *op, PyGC_Head *reachable)
 {
     if (PyObject_IS_GC(op)) {
         PyGC_Head *gc = AS_GC(op);
-        const Py_ssize_t gc_refs = _PyGCHead_REFS(gc);
+        const Py_ssize_t gc_refs = gc_get_refs(gc);
 
-        if (gc_refs == 0) {
-            /* This is in move_unreachable's 'young' list, but
-             * the traversal hasn't yet gotten to it.  All
-             * we need to do is tell move_unreachable that it's
-             * reachable.
-             */
-            _PyGCHead_SET_REFS(gc, 1);
+        if (gc->gc.gc_next == NULL || !gc_is_collecting(gc)) {
+            return 0;
         }
-        else if (gc_refs == GC_TENTATIVELY_UNREACHABLE) {
+        if (gc->gc.gc_prev & MASK_TENTATIVELY_UNREACHABLE) {
             /* This had gc_refs = 0 when move_unreachable got
              * to it, but turns out it's reachable after all.
              * Move it back to move_unreachable's 'young' list,
              * and move_unreachable will eventually get to it
              * again.
              */
+            gc->gc.gc_prev &= ~MASK_TENTATIVELY_UNREACHABLE;
             gc_list_move(gc, reachable);
-            _PyGCHead_SET_REFS(gc, 1);
+            gc_set_refs(gc, 1);
+        }
+        else if (gc_refs == 0) {
+            /* This is in move_unreachable's 'young' list, but
+             * the traversal hasn't yet gotten to it.  All
+             * we need to do is tell move_unreachable that it's
+             * reachable.
+             */
+            gc_set_refs(gc, 1);
         }
         /* Else there's nothing to do.
          * If gc_refs > 0, it must be in move_unreachable's 'young'
          * list, and move_unreachable will eventually get to it.
-         * If gc_refs == GC_REACHABLE, it's either in some other
-         * generation so we don't care about it, or move_unreachable
-         * already dealt with it.
-         * If gc_refs == GC_UNTRACKED, it must be ignored.
          */
-         else {
-            assert(gc_refs > 0
-                   || gc_refs == GC_REACHABLE
-                   || gc_refs == GC_UNTRACKED);
-         }
+        else {
+            assert(gc_refs > 0);
```

Comment:
Why flags are no longer tested?

Reply:
Because there are no `GC_REACHABLE` and `GC_UNTRACKED` now.
Instead, there are "early return" above.

```
+        if (gc->gc.gc_next == NULL || !gc_is_collecting(gc)) {
+            return 0
```

`gc_refs == GC_REACHABLE` meant the object is tracked but not in current gc_list.  It is `!gc_is_collecting(gc)` now.

`gc_refs == GC_UNTRACKED` meant the object is not tracked.  It is `gc->gc.gc_next == NULL` now.